### DISCUSSION TITLE
Diogenes Compatibility List

### DISCUSSION THREAD

**Discussion by jglim** [jglim/CaesarSuite#18] [General]
This page shows the existing compatibility status of Diogenes

### Tester

Github handle (if available) of the tester

### ECU

ECU Name as displayed in Diogenes

### Variant

ECU Variant name, optionally with variant ID in round brackets.

### Physical

Physical interface type (e.g. HSCAN, KWP2000)

### Protocol

Communication protocol (e.g. UDS, KW2C3PE)

### Hardware

Device connecting ECU to test computer, optionally with driver DLL name in round brackets.

### Connection

Test result: `OK` if physical connection is established (e.g. HSCAN) and raw bytes can be sent and received.

### Variant Detection, Session Switching

Test result: `OK` if Diogenes "Init Contact" successfully switches session and selects the correct ECU variant

### Variant Coding

Test result: `OK` if ECU variant coding was successfully changed without error, and remains after restart (hard reset or power disconnect/reconnect)

### Accepted by Vehicle

Test result: `OK` if variant coding changes are accepted by vehicle without any error. Comments can be included e.g. new DTC errors, limp mode, SCN issues etc.

### Fingerprint Type

Describes official software behavior when setting fingerprints for variant coding:

- None: No fingerprint field, or it is included in the variant-string e.g. VGSNAG2
- Constant: Official clients write a constant value e.g. `00 00 00 01`, and this value does not affect ECU operation
- Dynamic: Official clients generate a different value for each unique variant string (This will be a problem, seems to be present in CR4_NFZ)

## Diogenes Compatibility List

Tester | ECU | Variant | Physical | Protocol | Hardware | Connection | Variant Detection, Session Switching | Variant Coding | Accepted by Vehicle | Fingerprint Type
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
@jglim | CRD3 | CRD3_Diag_2131h (2131) | HSCAN | UDS | OpenPort2 (op20pt32.dll) | OK | OK | OK : VCD_CRD3_Implicit_Coding_Restricted | ? | Constant
@N0cynym | VGSNAG2 | VGS2_0101 (0101) | HSCAN | KW2C3PE | AVDI (AbrPT32.dll) | OK | ? | OK : VCD_Entwicklung_Variantenkodierung_VGS_72 | ? | None
@Feezex | EZS906 | NCV3_0301 (0301) | HSCAN | KW2C3PE | Scanmatik2 (smj2534.dll) | OK | OK | ? | ? | ?
@Nackapatz | IC_204 | IC_FX_204_Mid_Line_AJ11_D_Muster (?) | HSCAN | UDS | OpenPort2 (op20pt32.dll) | OK | OK | OK : VCD_Laenderkodierung_Detail | ✅ | ?
@Nackapatz | IC_204 | IC_FX_204_Mid_Line_AJ11_D_Muster (?) | HSCAN | UDS | JDiag2.0 Pro (JDIAG32.dll) | OK | OK | OK : VCD_Laenderkodierung_Detail | ✅ | ?
@Nackapatz | SCCM166 | BR166_Valeo_0x0508 (0508) | HSCAN | UDS | OpenPort2 (op20pt32.dll) | OK | [❌](https://github.com/jglim/CaesarSuite/discussions/18#discussioncomment-588711) | - | - | -
@Nackapatz | SCCM166 | BR166_Valeo_0x0508 (0508) | HSCAN | UDS | JDiag2.0 Pro (JDIAG32.dll) | OK | [❌](https://github.com/jglim/CaesarSuite/discussions/18#discussioncomment-588711) | - | - | -
@Nackapatz | IC172 | IC_MFA_AeJ18 (?) | HSCAN | UDS | OpenPort2 (op20pt32.dll) | OK | OK | OK : WVC_04_Laenderkodierung_SCN_Schreiben | ✅ | ?
@Nackapatz | IC172 | IC_MFA_AeJ18 (?) | HSCAN | UDS | JDiag2.0 Pro (JDIAG32.dll) | OK | OK | OK : WVC_04_Laenderkodierung_SCN_Schreiben | ✅ | ?
@jglim | IC_204 | IC212_IC_204_Mid_Line_AJ10 (1307) | HSCAN | UDS | OpenPort2 (op20pt32.dll) | OK | OK | OK : VCD_05_Variantenkodierung | ? | None

Please let me know in this thread if there should be any changes (or additions!) to this list.

**Comment by Nackapatz** [comment]
@Nackapatz/IC_204/IC_FX_204_Mid_Line_AJ11_D_Muster/HSCAN_UDS_500/UDS/OpenPort2 (op20pt32.dll)/OK/OK/OK:VCD_Laenderkodierung_Detail/Yes/?

@Nackapatz/IC_204/IC_FX_204_Mid_Line_AJ11_D_Muster/HSCAN_UDS_500/UDS/JDiag2.0 Pro JDIAG32.dll/OK/OK/OK:VCD_Laenderkodierung_Detail/Yes/?

@Nackapatz/SCCM166/BR166_Valeo_0x0508/HSCAN_UDS_500/UDS/JDiag2.0 Pro JDIAG32.dll/OK/
Session_Switching: "SES_Programming_P2_CAN_ECU_max: InBytes [22] length mismatch (expecting 2)"
SES_Default_P2_CAN_ECU_max_functional: "Retrying: empty buffer: BUFFER_EMPTY for request 10 81"
/No Coding/?/?

@Nackapatz/SCCM166/BR166_Valeo_0x0508/HSCAN_UDS_500/UDS/OpenPort2 (op20pt32.dll)/OK/
Session_Switching: "SES_Programming_P2_CAN_ECU_max: InBytes [22] length mismatch (expecting 2)"
SES_Default_P2_CAN_ECU_max_functional: "Retrying: empty buffer: BUFFER_EMPTY for request 10 81"
/No Coding/?/?

@Nackapatz/IC172/IC_MFA_AeJ18/HSCAN_UDS_500/UDS/OpenPort2 (op20pt32.dll)/OK/OK/
WVC_04_Laenderkodierung_SCN_Schreiben
[!] Retrying: empty buffer: BUFFER_EMPTY for request 2E 02 00 80 00 0A 00 10 17
VC Write completed
Coding saved
/Yes/?


@Nackapatz/IC172/IC_MFA_AeJ18/HSCAN_UDS_500/UDS/JDiag2.0 Pro JDIAG32.dll/OK/OK/
WVC_04_Laenderkodierung_SCN_Schreiben
[!] Retrying: empty buffer: BUFFER_EMPTY for request 2E 02 00 80 00 0A 00 10 17
VC Write completed
Coding saved
/Yes/?

**Reply by jglim** [reply]
Added, thank you @Nackapatz !

**Reply by Nackapatz** [reply]
Updated. Will try some other units too.

**Comment by diegohmarciano** [comment]
Hi dears, I have a bench setup as I'm working with Ghidra on ME9.7, having issues to actually unlock 0x23 to simplify my life to read some ram values. Anyways, last week I was working with an Adaptive Brake unit, ABR2XT, for whatever reason it was impossible to connect with CaesarSuite, vediamo worked find with the same CBF, I thought it was maybe me as it was the first time to actually use CaesarSuite to work on an ECU, but been using it on other (IC_204, ME97) without issues.
I was able to do what I needed so far with vediamo (but wasn't able to flash the unit which was an experiment I wanted to do as I have this unit on my hands just for experimentation purposes along with another SAMF that I have just for experimentation/bench work), is it worth that I collect any data on this ABR2XT unit to collaborate?

**Reply by jglim** [reply]
Hi Diego, thanks for using CaesarSuite and for writing in about this issue. More data is always appreciated :^)

If you don't mind the trouble, the best way to capture logs is using J2534-Shim. Setup instructions are [available here](https://github.com/jglim/CaesarSuite/discussions/11). The logs are handy as we can see where our behavior diverges from Vediamo.

There's also a v2 branch which may or may not behave better: https://github.com/jglim/CaesarSuite/tree/v2 , and while there isn't proper builds right now, you can try a random build [from this thread](https://github.com/jglim/CaesarSuite/issues/56#issuecomment-1500454932).

**Reply by diegohmarciano** [reply]
Sorry bud, I was just reading something here and forgot about this chat, the reason I missed to capture is because this piece was borrowed, and short after I posted these they needed the piece, so I just did it with my other tools (vediamo / AVDI), but will try to participate on another area I've been playing for a long time without success
